pandas/core/indexes/period.py

```diff
@@ -674,6 +674,15 @@ def to_timestamp(self, freq=None, how='start'):
         """
         how = _validate_end_alias(how)
 
+        end = how == 'E'
+        if end:
+            if freq == 'B':
```

Comment:
For some reason ``freq='B'`` had to be handled differently because it was causing a lot of resample tests to fail when handled in the same way as all the other freq values. Not sure if this is a bug or if there is a better way of handling this.